hydra/_internal/hydra.py

```diff
@@ -141,37 +142,28 @@ def multirun(
         return ret
 
     @staticmethod
-    def get_sanitized_hydra_cfg(src_cfg: DictConfig) -> DictConfig:
-        cfg = copy.deepcopy(src_cfg)
+    def sanitize_hydra_cfg(cfg: DictConfig) -> None:
+        """
+        Mutate the input cfg:
+          - delete all toplevel keys except "hydra".
+          - delete `hydra.help` and `hydra.hydra_help`.
+        """
```

Comment:
https://github.com/facebookresearch/hydra/pull/1696/commits/57452d5b9e71485cb3e92d6525f2d71a2cb3abc5 is a refactoring:
  - `get_sanitized_hydra_cfg` -> `sanitize_hydra_cfg`
  -  `get_sanitized_cfg` -> `sanitize_cfg`


Before, these methods were making a deepcopy in some branches, and not making a deepcopy in other branches.
Now, they never make a deepcopy (i.e. they mutate their input directly). This refactoring should resolve the LGTM alert that was posted in the PR's main thread..

Reply:
Looks like get_sanitized_hydra_cfg was always copying and get_sanitized_cfg was composing (so no need to copy).
Not sure what alerts you are referring to.
```python
def get_sanitized_hydra_cfg(src_cfg: DictConfig) -> DictConfig:
        cfg = copy.deepcopy(src_cfg)
```

Reply:
Looking at the warning, the refactoring sounds like way too big a hammer.

![image](https://user-images.githubusercontent.com/376455/124199795-a135d800-da88-11eb-9b0b-7bef0de08253.png)

Reply:
Fair enough. I reverted the refactoring in https://github.com/facebookresearch/hydra/pull/1696/commits/f0fa6cdf104fb3a93a9e9a220f040ea02e8892b3 and then did a simpler fix in https://github.com/facebookresearch/hydra/pull/1696/commits/06258d203ae51cba45659b6e4eb3bb8b7a8e71fd (calling `get_sanitized_cfg` earlier).